packages/task/src/browser/task-service.ts

```diff
@@ -401,15 +418,30 @@ export class TaskService implements TaskConfigurationClient {
         }
     }
 
-    async runTaskByLabel(taskLabel: string): Promise<boolean> {
+    async runTaskByLabel(taskLabel: string): Promise<TaskInfo | undefined> {
         const tasks: TaskConfiguration[] = await this.getTasks();
         for (const task of tasks) {
             if (task.label === taskLabel) {
-                await this.runTask(task);
-                return true;
+                return this.runTask(task);
             }
         }
-        return false;
+
+        return;
+    }
+
+    async runWorkspaceTask(workspaceFolderUri: string | undefined, taskName: string): Promise<TaskInfo | undefined> {
```

Comment:
I tested in my local env and it didn't work. Please see GIF below:

![Peek 2019-10-07 22-38](https://user-images.githubusercontent.com/37082801/66363171-53437d00-e953-11e9-8c2f-76d395dbeeab.gif)

I checked the debugger. Looks like the `taskName` passed into this function was an object, not a string.